deepchecks/checks/performance/model_error_analysis.py

```diff
@@ -0,0 +1,317 @@
+            min_feature_contribution: float = 0.15,
+            min_error_model_score: float = 0.5,
+            min_segment_size: float = 0.05,
+            alternative_scorer: Dict[str, Callable] = None,
+            n_samples: int = 50_000,
+            n_display_samples: int = 5_000,
+            random_seed: int = 42
+    ):
+        super().__init__()
+        self.max_features = max_features
+        self.min_feature_contribution = min_feature_contribution
+        self.min_error_model_score = min_error_model_score
+        self.min_segment_size = min_segment_size
+        if (alternative_scorer is not None) and (len(alternative_scorer) > 1):
+            raise DeepchecksProcessError(
+                'Only one alternative scorer is allowed.'
+            )
+        self.alternative_scorer = initialize_multi_scorers(alternative_scorer)
+        self.n_samples = n_samples
+        self.n_display_samples = n_display_samples
+        self.random_state = random_seed
+
+        self._scorer_name = None
+
+    def run(self, train_dataset: Dataset, test_dataset: Dataset, model=None) -> CheckResult:
+        """Run check.
+
+        Args:
+            train_dataset (Dataset): The training dataset object. Must contain a label.
+            test_dataset (Dataset): The test dataset object. Must contain a label.
+            model (BaseEstimator): A scikit-learn-compatible fitted estimator instance.
+        """
+        # Validations
+        Dataset.validate_dataset(train_dataset)
+        Dataset.validate_dataset(test_dataset)
+        train_dataset.validate_label()
+        test_dataset.validate_label()
+        train_dataset.validate_shared_label(test_dataset)
+        train_dataset.validate_shared_features(test_dataset)
+        train_dataset.validate_shared_categorical_features(test_dataset)
+        validate_model(train_dataset, model)
+
+        task_type = task_type_check(model, train_dataset)
+
+        # If user defined scorers used them, else use a single scorer
+        if self.alternative_scorer:
+            scorer = get_scorers_list(model, train_dataset, self.alternative_scorer, multiclass_avg=True)[0]
+        else:
+            scorer = get_scorer_single(model, train_dataset, multiclass_avg=True)
+
+        cat_features = train_dataset.cat_features
+
+        train_dataset = train_dataset.sample(self.n_samples, random_state=self.random_state)
+        test_dataset = test_dataset.sample(self.n_samples, random_state=self.random_state)
+
+        # Create scoring function, used to calculate the per sample model error
+        if task_type == ModelType.REGRESSION:
+            def scoring_func(dataset):
+                return per_sample_mse(dataset.label_col, model.predict(dataset.features_columns))
+        else:
+            def scoring_func(dataset):
+                return per_sample_binary_cross_entropy(dataset.label_col,
+                                                       model.predict_proba(dataset.features_columns))
```

Comment:
Why complicate this so much? Why create a scoring_func function in an if-else condition, if its only a simple wrapper, and its only used in the next lines of code?

Reply:
It's simple but it's better than just doing it all in the if else IMO

Reply:
Disagree, no need to create an instance of a wrapper if its only used once and directly afterwards.
But not going to fight on it. Will change it behind your back sometime

Reply:
Used twice!
```
train_scores = scoring_func(train_dataset)
test_scores = scoring_func(test_dataset)
```